src/ff1.rs

```diff
@@ -90,6 +98,54 @@ impl RadixOps<u16> for u16 {
     }
 }
 
+/// A radix 2^i for i in [1..16). It does not use floating-point arithmetic.
+#[derive(Debug, PartialEq)]
+pub struct PowerTwoRadix {
+    radix: u16,
+    log_radix: u8,
+}
+
+impl PowerTwoRadix {
+    pub fn from(radix: u16) -> Result<Self, ()> {
+        let mut tmp = radix;
+        let mut log_radix = 0;
+        let mut found = false;
+        for i in 0..16 {
+            if tmp & 1 != 0 {
+                // Only a single bit should be set
+                if found {
+                    return Err(());
+                }
+                log_radix = i;
+                found = true;
+            }
+            tmp >>= 1;
+        }
```

Comment:
A simpler way of testing for a power of 2 is ``radix != 0 && (radix & (radix - 1)) == 0``, but I guess that doesn't give you ``log_radix``.

Reply:
I do something similar in `BinaryNumeralString`, but yeah, here I need to know which bit of the radix is set, and there aren't any existing methods for that.